### ISSUE TITLE
The perl "read" function can continue to read after EOF

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#4014]
Migrated from [rt.perl.org#7024](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=7024) (status was 'open')

Searchable as RT7024$

**Comment by p5pRT** [comment]
### From mcba@phys.unsw.edu.au

This bug report refers to the latest perl found on Redhat 7\.1 Intel Linux
systems\.

The bug occurs when I try to read data from a CCD camera device driver
that I wrote\. The driver returns 0 at end\-of\-file to signal that the image
has been completed\, BUT if you then try to read again\, it will return
EWOULDBLOCK since it assumes that you have re\-opened the device and are
expecting a new image\. The perl "read" function sometimes ignores the 0\,
and appears to try reading from the device again\.

Now\, this is arguably a bug in my device driver \(and I will fix it\)\,
however\, the fundamental point remains that the C "read" function returned
zero\, indicating EOF\, and the perl "read" function can ignore this\. Perl
"read" seems to assume that if an EOF is returned once\, then it should be
returned on subsequent reads\.

Note that the perl "sysread" function works fine with my driver
\(as do the usual UNIX utilities such as "cp" and "cat"\)\.

I hope that this bug report is useful\. Sorry I didn't use "perlbug"\.

Best regards\,
Michael

**Comment by tonycoz** [comment]
There's no example code here, nor any information about which version of perl was used.

The comment does assume you can't continue to read after reaching EOF, but that is untrue for a file - if you read a file to EOF and that file is extended, the next read() system call shouldn't return EOF.

Given the age and lack of specifics I think this is closable.

**Comment by jkeenan** [comment]
> There's no example code here, nor any information about which version of perl was used.
> 
> The comment does assume you can't continue to read after reaching EOF, but that is untrue for a file - if you read a file to EOF and that file is extended, the next read() system call shouldn't return EOF.
> 
> Given the age and lack of specifics I think this is closable.

Agreed. Give a couple of days for further comments, then proceed.  Thanks.

**Comment by jkeenan** [comment]
> > There's no example code here, nor any information about which version of perl was used.
> > The comment does assume you can't continue to read after reaching EOF, but that is untrue for a file - if you read a file to EOF and that file is extended, the next read() system call shouldn't return EOF.
> > Given the age and lack of specifics I think this is closable.
> 
> Agreed. Give a couple of days for further comments, then proceed. Thanks.

Closing as per schedule.
